burnman/eos/hp.py

```diff
@@ -153,7 +153,12 @@ def gibbs_free_energy(self,pressure,temperature, volume, params):
             else:
                 Gdisord=0.0
 
-        return params['H_0'] + self.__intCpdT(temperature, params) - temperature*(params['S_0'] + self.__intCpoverTdT(temperature, params)) + intVdP + Gdisord
+        if params.has_key('magnetic_moment'):
+            Gmagnetic=self._magnetic_gibbs(pressure, temperature, params)
+        else:
+            Gmagnetic=0.0
+
+        return params['H_0'] + self.__intCpdT(temperature, params) - temperature*(params['S_0'] + self.__intCpoverTdT(temperature, params)) + intVdP + Gdisord + Gmagnetic
```

Comment:
Oooh, a philosophical question :)
You mean setting the magnetic params to zero if they don't exist? Would that help much? 

In this case, magnetic_moment is just a cue to compute the magnetic gibbs contribution if necessary (i.e. if the phase is magnetic). If "magnetic_moment" was set to zero in validate_params, then the gibbs contribution would be zero, but running the computations would probably be a lot more expensive than running the param check.